packages/shared/src/hooks/post/useCommentEdit.ts

```diff
@@ -1,41 +1,33 @@
-import { useQuery } from '@tanstack/react-query';
 import { useMemo, useState } from 'react';
-import { Comment, COMMENT_BY_ID_QUERY } from '../../graphql/comments';
-import { generateQueryKey, RequestKey } from '../../lib/query';
-import { graphqlUrl } from '../../lib/config';
-import { useAuthContext } from '../../contexts/AuthContext';
-import { useRequestProtocol } from '../useRequestProtocol';
 import { CommentWrite, CommentWriteProps } from './common';
+import useCommentById from '../comments/useCommentById';
 
 interface UseCommentEdit extends CommentWrite {
   onEdit: (params: CommentWriteProps) => void;
 }
 
 export const useCommentEdit = (): UseCommentEdit => {
   const [state, setState] = useState<CommentWriteProps>();
-  const { commentId: id, lastUpdatedAt } = state ?? {};
-  const { user } = useAuthContext();
-  const { requestMethod } = useRequestProtocol();
-  const { data } = useQuery<{ comment: Comment }>(
-    generateQueryKey(RequestKey.Comment, user, id, lastUpdatedAt),
-    () => requestMethod(graphqlUrl, COMMENT_BY_ID_QUERY, { id }),
-    { enabled: !!id },
-  );
+  const { commentId: id } = state ?? {};
+  const { comment, onEdit } = useCommentById({ id });
 
   const inputProps = useMemo(() => {
-    if (!state || !data) {
+    if (!state || !comment) {
       return null;
     }
 
     return {
       editCommentId: state.commentId,
       parentCommentId: state.parentCommentId,
-      initialContent: data?.comment?.content,
+      initialContent: comment?.content,
     };
-  }, [data, state]);
+  }, [comment, state]);
 
   return {
-    onEdit: setState,
+    onEdit: (params) => {
```

Comment:
Let's add a `useCallback` here to make the function stable. Might get used somewhere as a dependency.